### PR TITLE
chore: remove Dagger from `HandleWorkflow`

### LINKED ISSUE #14030: Replace `.handle` Dagger subcomponents with records
### Problem

Dagger components are too heavyweight for the most performance critical paths.

### Solution

Replace them with `records`.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
- Closes #14030 by replacing `{Child,User}DispatchComponent`s with a `RecordDispatch` and replacing `UserTxnComponent` with a `UserTxn` record.
- Consolidates multiple smaller classes into private methods of `HandleWorkflow`.
- Moves the most self-contained "steps" into a `c.h.s.a.workflows.handle.steps` package.
- Adds more docs to the [key `Dispatch` abstraction](https://github.com/hashgraph/hedera-services/blob/14030-remove-dagger-from-handle-workflow/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/Dispatch.java#L43) as the fundamental unit of work in the `HandleWorkflow`.
- Unit test coverage will be restored by moving _test-clients/_ into a `hapi` module under `hedera-app` tests and running all `@HapiTest`s in embedded mode in CI.

### CODE REVIEW

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java [hunk 1]
```diff
@@ -1124,23 +1126,30 @@ private void writeHashes(final long maxValidPath, final Stream<VirtualHashRecord
...
-            // nothing to do
+        // If both streams are empty, no new data files should be created. One simple way to
+        // check emptiness is to use iterators. The streams aren't processed in parallel anyway
+        final Iterator<VirtualLeafRecord<K, V>> dirtyIterator = dirtyLeaves
+                .sorted(Comparator.comparingLong(VirtualLeafRecord::getPath))
+                .iterator();
+        final Iterator<VirtualLeafRecord<K, V>> deletedIterator = deletedLeaves.iterator();
+
+        if (!dirtyIterator.hasNext() && !deletedIterator.hasNext()) {
+            // Nothing to do
```
Comment (anthony-swirldslabs) [inline platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java:1141]: It's odd that GitHub shows these changes here as they belong to a separate PR/commit ( https://github.com/hashgraph/hedera-services/pull/14020 ). Looks like a bug in GitHub.

You may want to pull latest changes from develop into your branch and update your PR to try and fix it. This should hopefully remove `platform-data` from the list of reviewers.

Comment (tinker-michaelj) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java]: Hmm that was confusing, hopefully good now. 🤔

Comment (anthony-swirldslabs) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java]: Thanks for fixing this! There's no longer any files owned by platform-data, so you don't need our review on your PR anymore.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! Thanks @tinker-michaelj . As discussed, once PCLI is fixes we will validte all records by diff testing.
